src/kernel/mempool_entry.h

```diff
@@ -191,6 +191,10 @@ struct NewMempoolTransactionInfo {
      */
     int64_t m_virtual_transaction_size;
     unsigned int txHeight;
+    bool m_from_disconnected_block;
+    bool m_submitted_in_package;
+    bool m_chainstate_is_current;
+    bool m_has_no_mempool_parents;
```

Comment:
9b0843042ab5c30e6ef4753d590c69c4811d9a70 seems to be doing multiple things: (1) making `CBlockPolicyEstimator` react to validationinterface and (2) delegating the `validForFeeEstimation` from validation to the `CBlockPolicyEstimator` by providing these bools in the `TransactionAddedToMempool` signal (but this isn't fully done until the next commit which cleans things up).

These should be separate commits, e.g. (2) and then (1).

Reply:
Thank you

This is fixed.
1.  a469d937dbbadedeb58e0d5d1c44ed7881a7ebb9 introduces `NewMempoolTransactionInfo` with all the necessary fields and updates `TransactionAddedToMempool` callback parameter.
2. ac82f58e45421aa5aec1d3e6e992d1c369472a10 updates the fee estimator from `CValidationInterface` notifications and manage delegating `validForFeeEstimation`  to `CBlockPolicyEstimator::processTransaction` together the cleanups.